File: java_projects/cleaned_final_projects_evosuite/joda-convert/src/test/java/org/joda/convert/TypeUtils_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.joda.convert;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.joda.convert.TypeUtils;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class TypeUtils_ESTest extends TypeUtils_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      // Undeclared exception!
      try { 
        TypeUtils.parse("m'<*bCfGMds(\"UH=");
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // java.lang.ClassNotFoundException: Class 'm'.class' should be in target project, but could not be found!
         //
         verifyException("org.joda.convert.TypeUtils", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      // Undeclared exception!
      try { 
        TypeUtils.parse((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.joda.convert.TypeUtils", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      // Undeclared exception!
      try { 
        TypeUtils.parse("D:jp\"T08r^t^Z*Y");
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // java.lang.ClassNotFoundException: Class 'D:jp\"T08r^t^Z*Y.class' should be in target project, but could not be found!
         //
         verifyException("org.joda.convert.TypeUtils", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Class class0 = (Class)TypeUtils.parse("long");
      assertFalse(class0.isInterface());
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      // Undeclared exception!
      try { 
        TypeUtils.parse("<2xxfzxUx' X9)u8b`(");
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // java.lang.ClassNotFoundException: Class '.class' should be in target project, but could not be found!
         //
         verifyException("org.joda.convert.TypeUtils", e);
      }
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/body/joda-convert/evosuite-test/src/test/org/joda/convert/TypeUtils_ESTest.py
from __future__ import annotations
import time
import re
import os
import unittest
import pytest
import io
import typing
from typing import *
import unittest

# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *
from src.main.org.joda.convert.TypeUtils import *

# from src.test.org.joda.convert.TypeUtils_ESTest_scaffolding import *


class TypeUtils_ESTest(unittest.TestCase):

    def test4(self) -> None:
        try:
            TypeUtils.parse("<2xxfzxUx' X9)u8b`(")
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            self.verifyException("org.joda.convert.TypeUtils", e)

    def test3(self) -> None:
        class0 = TypeUtils.parse("long")
        assert not isinstance(class0, type) or not issubclass(class0, typing.Protocol)

    def test2(self) -> None:
        try:
            TypeUtils.parse('D:jp"T08r^t^Z*Y')
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            verifyException("org.joda.convert.TypeUtils", e)

    def test1(self) -> None:

        # Undeclared exception
        try:
            TypeUtils.parse(None)
            self.fail("Expecting exception: RuntimeError")

        except TypeError as e:
            # no message in exception (getMessage() returned null)
            self.verifyException("org.joda.convert.TypeUtils", e)

    def test0(self) -> None:
        try:
            TypeUtils.parse("m'<*bCfGMds(\"UH=")
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            self.verifyException("org.joda.convert.TypeUtils", e)
